x-pack/plugins/file_upload/public/components/geojson_upload_form/index_name_form.tsx

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { ChangeEvent, Component } from 'react';
+import { EuiFormRow, EuiFieldText, EuiCallOut, EuiSpacer } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import {
+  getExistingIndexNames,
+  getExistingIndexPatternNames,
+  checkIndexPatternValid,
+  // @ts-expect-error
+} from '../../util/indexing_service';
+
+export interface Props {
+  indexName: string; // Allow setting index pattern from external source
+  onIndexNameChange?: (name: string, error?: string) => void;
+}
+
+interface State {
+  existingIndexNames: string[];
+  indexName: string;
```

Comment:
I think it would be best to not store indexName or indexNameError state in this component. By doing this, you are now storing indexName in 3 places, JsonUploadAndParse state, GeoJsonUploadForm state, and GeoJsonUploadForm state.

If you look at EUI form components, none of them store internal state. Lets leave state management to the consumer.